var/spack/repos/builtin/packages/openloops/package.py

```diff
@@ -274,12 +274,16 @@ def configure(self, spec, prefix):
             f.write("fortran_compiler = {0}\n".format(env["SPACK_FC"]))
             if self.spec.satisfies("@1.3.1") and not is_intel:
                 f.write("gfortran_f_flags = -ffree-line-length-none\n")
-            if self.spec.satisfies("@2.1.1:") and not is_intel:
+            if self.spec.satisfies("@2.1.1") and not is_intel:
                 f.write("gfortran_f_flags = -ffree-line-length-none " + "-fdollar-ok ")
                 if self.spec.target.family == "aarch64":
                     f.write("-mcmodel=small\n")
                 else:
                     f.write("-mcmodel=medium\n")
+            if self.spec.satisfies("@2.1.2:") and not is_intel:
+                f.write("gfortran_f_flags = -ffree-line-length-none " + "-fdollar-ok\n")
+                if self.spec.target.family == "aarch64":
+                    f.write("cmodel = small\n")
```

Comment:
It seems the recommendation on the linked gitlab issue is to use large, not small.

Reply:
The `-mcmodel=large` does not seem to build though as it is not compatible with `-fPIC`. I believe is added automatically by openloops for gfortran ([here](https://gitlab.com/openloops/OpenLoops/-/blob/OpenLoops-2.1.2/scons-local/scons-local-3.0.5/SCons/Tool/gfortran.py?ref_type=tags), but no 100% sure as I didn't trace it) and I don't see a way to change it.

```
scons: Building targets ...
/usr/bin/gfortran -o lib_src/olcommon/obj/kind_types.os -c -ffree-line-length-none -fdollar-ok -mcmodel=large -O2 -fPIC -Ilib_src/olcommon/mod -Jlib_src/olcommon/mod lib_src/olcommon/obj/kind_types.f90
f951: sorry, unimplemented: code model 'large' with '-fPIC'
scons: *** [lib_src/olcommon/obj/kind_types.os] Error 1
scons: building terminated because of errors.
```

Apologies for not communicating that my last push was untested. Computer access during the holidays has been rather sporadic.

Shall I open a new PR changing the cmodel back to small?

Reply:
> Shall I open a new PR changing the cmodel back to small?

Ah, sorry about merging this before testing (I don't regularly test arm builds). Yes, please open a new PR.